src/Weather/Skysight/SkysightAPI.hpp

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright_License {
+
+  XCSoar Glide Computer - http://www.xcsoar.org/
+  Copyright (C) 2000-2016 The XCSoar Project
+  A detailed list of copyright holders can be found in the file "AUTHORS".
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+#ifndef WEATHER_SKYSIGHTAPI_HPP
+#define WEATHER_SKYSIGHTAPI_HPP
+
+#include "Metrics.hpp"
+#include "Util/tstring.hpp"
+
+class SkysightAPI final {
+public:
+  SkysightAPI() {
+  };
+  ~SkysightAPI();
+  
+  bool IsInited();
+  
+  tstring region;
+  
+  std::map<tstring, tstring> regions;
+  
+  void Init(const tstring email, const tstring password,
```

Comment:
Why an Init method? The constructor should initialize the object.

Reply:
Move the code in Init() to the constructor. As a side effect,
-  SkysightAPI api;
+  SkysightAPI *api;